integrations/lib/multiwoven/integrations/source/aws_athena/client.rb

```diff
@@ -0,0 +1,121 @@
+# frozen_string_literal: true
+
+require "aws-sdk-athena"
+
+module Multiwoven::Integrations::Source
+  module AWSAthena
+    include Multiwoven::Integrations::Core
+    class Client < SourceConnector
+      def check_connection(connection_config)
+        connection_config = connection_config.with_indifferent_access
+        athena_client = create_connection(connection_config)
+        athena_client.list_work_groups
+        ConnectionStatus.new(status: ConnectionStatusType["succeeded"]).to_multiwoven_message
+      rescue StandardError => e
+        ConnectionStatus.new(status: ConnectionStatusType["failed"], message: e.message).to_multiwoven_message
+      end
+
+      def discover(connection_config)
+        connection_config = connection_config.with_indifferent_access
+        query = "SELECT table_name, column_name, data_type, is_nullable FROM information_schema.columns WHERE table_schema = '#{connection_config[:schema]}' ORDER BY table_name, ordinal_position;"
+        db = create_connection(connection_config)
+        response = db.start_query_execution(
```

Comment:
We can make this DRY by extracting this into a function as this is used in read too. This will also increase the readability of the code